src/core/utils/fs.cpp

```diff
@@ -516,3 +521,61 @@ QString Utils::Fs::cacheLocation()
         locationDir.mkpath(locationDir.absolutePath());
     return location;
 }
+
+#ifndef DISABLE_GUI
+// Open the given path with an appropriate application
+void Utils::Fs::openPath(const QString& absolutePath)
+{
+    const QString path = fromNativePath(absolutePath);
+    // Hack to access samba shares with QDesktopServices::openUrl
+    if (path.startsWith("//"))
+        QDesktopServices::openUrl(toNativePath("file:" + path));
+    else
+        QDesktopServices::openUrl(QUrl::fromLocalFile(path));
+}
+
+// Open the parent directory of the given path with a file manager and select
+// (if possible) the item at the given path
+void Utils::Fs::openFolderSelect(const QString& absolutePath)
+{
+    const QString path = fromNativePath(absolutePath);
+#ifdef Q_OS_WIN
+    if (QFileInfo(path).exists()) {
+        // Syntax is: explorer /select, "C:\Folder1\Folder2\file_to_select"
+        // Dir separators MUST be win-style slashes
+        QProcess::startDetached("explorer.exe /select,\"" + toNativePath(path) + "\"");
```

Comment:
Yes, the path must be completely whitespace-free, that's why I was using C:\. Sorry for not being clear on this.

The other method doesn't work because Qt will put quotes around your quotes if I remember correctly.